### PR TITLE
EPN subtype method added

### PR DESCRIPTION
**Pull request by runjin326**
### Purpose

Add method section for EPN subtyping

### Issue
https://github.com/AlexsLemonade/OpenPBTA-manuscript/issues/136

#### Which areas should receive a particularly close look?
1). When I was reading the [README.md in the analysis repo](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/molecular-subtyping-EPN/README.md#folder-content), it looks like the table for exclusive fusions were not particularly correct: 
image
Should it be:
`C11orf95--RELA`, `LTBP3--RELA` for RELA 
`C11orf95--YAP1`, `YAP1--MAMLD1` and `YAP1--FAM118B` for `YAP1`

2). Please check to see whether the description is too detailed. 

#### Is there anything that you want to discuss further?
No.

#### Is the pull request ready for review?
Yes.

### Pull review checklist

Unless otherwise noted above, this PR will be considered ready for review when all four items have been checked.

- [x] All changes to text follow "one sentence per line" [[Manubot instructions](https://github.com/AlexsLemonade/OpenPBTA-manuscript/blob/master/USAGE.md#manubot-markdown)]
- [x] All citations follow the [Manubot citation instructions](https://github.com/AlexsLemonade/OpenPBTA-manuscript/blob/master/USAGE.md#citations)
- [ ] All changes or additions to tables follow the [Manubot table instructions](https://github.com/AlexsLemonade/OpenPBTA-manuscript/blob/master/USAGE.md#tables)
- [ ] All figures follow the [Manubot figure instructions](https://github.com/AlexsLemonade/OpenPBTA-manuscript/blob/master/USAGE.md#figures)
- [ ] There are no new spelling errors identified by the [Manubot spellchecker](https://github.com/AlexsLemonade/OpenPBTA-manuscript/blob/master/USAGE.md#spellchecking)

### Spellcheck Step

The dictionary used for spellchecking can be updated.
Edit the file in `build/assets/custom-dictionary.txt` by adding new entries to the end.
You do not need to change anything else.
However, if you want to update the first line to have an accurate count of words and you want to remove non-unique ones, run the following command from within `build/assets` on your favorite OS X or Linux machine:
```
(( len = $(awk '!a[$0]++' < custom-dictionary.txt | wc -l ) - 1 )); tmpfile="$(mktemp)"; echo "personal_ws-1.1 en $len utf-8" > $tmpfile; tail -n +2 custom-dictionary.txt | awk '!a[$0]++' >> $tmpfile; mv $tmpfile custom-dictionary.txt
```

### CODE REVIEW

**Comment by jaclyn-taroni** [comment]
That table looks correct to me based on the original issue about EPN subtyping: https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/245

**Comment by runjin326** [comment]
@jaclyn-taroni, thanks for reviewing this. If they are correct, then I am a bit confused - for `PTEN--TAS2R1` and `C11orf95--MAML2`, since they are prioritized and samples with such fusions would exclusively be one type but on line 89 of README.md specifying non-exclusive molecular characterization, it said `This  table contains featured that were _not_ prioritized meaning if a sample has `PTEN--TAS2R1` and `C11orf95--MAML2` fusions, that sample will show as subgrouped under both  `EPN, ST RELA`  and `EPN, ST YAP1`. And I think the code also took these two fusions as non-prioritized?

**Comment by jaclyn-taroni** [comment]
Oh I see - the question is about the prioritized column specifically. I am planning on reviewing this more thoroughly this afternoon and will keep this in mind.

File: content/03.methods.md [hunk 1]
```diff
@@ -388,6 +388,38 @@ Additional details can be found in the analysis [notebook](https://alexslemonade
...
+2.  Any tumor with fusions containing `YAP1` as fusion partner, such as `C11orf95--YAP1`, `YAP1--MAMLD1` and `YAP1--FAM118B`, was subtyped as `EPN, ST YAP1`.
+3. Any tumor with the following molecular characterization would be subtyped as `EPN, PF A`: 
+- _CXorf67_ expression z-score of over 3
+- _CXorf67_ expression z-score of over 3 and 1q gain 
+- _TKTL1_ expression z-score of over 3 and 1q gain
+4. Any tumor with the following molecular characterization would be subtyped as `EPN, PF B`:
+- _GPBP17_ expression z-score of over 3 and chromosome 6 long arm (6q) loss
+- _GPBP17_ expression z-score of over 3 and chromosome 6 short arm (6p) loss
+- _IFT46_ expression z-score of over 3 and chromosome 6 long arm (6q) loss
+- _IFT46_ expression z-score of over 3 and chromosome 6 short arm (6p) loss
```
Comment (jaclyn-taroni) [inline content/03.methods.md:403]:
```suggestion
- _IFT46_ expression z-score of over 3 and loss of 6q or 6p
```

File: content/03.methods.md [hunk 2]
```diff
@@ -388,6 +388,38 @@ Additional details can be found in the analysis [notebook](https://alexslemonade
 6. Non-MB and non-ATRT embryonal tumors with <i>CIC-NUTM1</i> or other <i>CIC</i> fusions, were subtyped as `CNS EFT-CIC` (CNS Ewing sarcoma family tumor with <i>CIC</i> alteration).
 7. Non-MB and non-ATRT embryonal tumors that did not fit any of the above categories were subtyped as CNS Embryonal, NOS (CNS Embryonal tumor, not otherwise specified).
 
+Ependymoma (EPN) were subtyped into `EPN, ST RELA`, `EPN, ST YAP1`, `EPN, PF A` and `EPN, PF B` based on evidence for these molecular subgroups as described in Pajtler et al. [@doi:10.1016/j.ccell.2015.04.002]. 
+Briefly, fusion, CNV and gene expression data were used to subtype EPN as followed:
+1. Any tumor with fusions containing `RELA` as fusion partner, e.g., `C11orf95--RELA`, `LTBP3--RELA`, was subtyped as `EPN, ST RELA`.
+2.  Any tumor with fusions containing `YAP1` as fusion partner, such as `C11orf95--YAP1`, `YAP1--MAMLD1` and `YAP1--FAM118B`, was subtyped as `EPN, ST YAP1`.
```
Comment (jaclyn-taroni) [inline content/03.methods.md:394]:
```suggestion
2. Any tumor with fusions containing `YAP1` as fusion partner, such as `C11orf95--YAP1`, `YAP1--MAMLD1` and `YAP1--FAM118B`, was subtyped as `EPN, ST YAP1`.
```

File: content/03.methods.md [hunk 3]
```diff
@@ -388,6 +388,38 @@ Additional details can be found in the analysis [notebook](https://alexslemonade
...
+3. Any tumor with the following molecular characterization would be subtyped as `EPN, PF A`: 
+- _CXorf67_ expression z-score of over 3
+- _CXorf67_ expression z-score of over 3 and 1q gain 
+- _TKTL1_ expression z-score of over 3 and 1q gain
+4. Any tumor with the following molecular characterization would be subtyped as `EPN, PF B`:
+- _GPBP17_ expression z-score of over 3 and chromosome 6 long arm (6q) loss
+- _GPBP17_ expression z-score of over 3 and chromosome 6 short arm (6p) loss
+- _IFT46_ expression z-score of over 3 and chromosome 6 long arm (6q) loss
+- _IFT46_ expression z-score of over 3 and chromosome 6 short arm (6p) loss
+Any tumor with the above molecular characteristics would be exclusively subtyped to the designated group.
```
Comment (jaclyn-taroni) [inline content/03.methods.md:404]:
```suggestion

Any tumor with the above molecular characteristics would be exclusively subtyped to the designated group. 
```

Extra line here helps with readability in the rich diff.

File: content/03.methods.md [hunk 4]
```diff
@@ -388,6 +388,38 @@ Additional details can be found in the analysis [notebook](https://alexslemonade
...
+Any tumor with the above molecular characteristics would be exclusively subtyped to the designated group. 
+
+For all other remaining EPN tumors without above molecular characteristics, they would be subtyped to `EPN, ST RELA` and `EPN, ST YAP1` in a non-exclusive way if any of the following alterations were present.
+1. Any tumor with the following alterations would have `EPN, ST RELA` assigned: 
+- `PTEN--TAS2R1` fusion 
+- chromosome 9 short arm (9p) loss 
+- chromosome 9 long arm (9q) loss 
+- _RELA_ expression z-score of over 3 
+- _L1CAM_ expression z-score of over 3 
+2. Any tumor with the following alterations would have `EPN, ST YAP1` assigned:
```
Comment (jaclyn-taroni) [inline content/03.methods.md:413]:
```suggestion
2. Any tumor with the following alterations was assigned `EPN, ST YAP1`: 
```

File: content/03.methods.md [hunk 5]
```diff
@@ -388,6 +388,38 @@ Additional details can be found in the analysis [notebook](https://alexslemonade
...
+- _TKTL1_ expression z-score of over 3 and 1q gain
+4. Any tumor with the following molecular characterization would be subtyped as `EPN, PF B`:
+- _GPBP17_ expression z-score of over 3 and chromosome 6 long arm (6q) loss
+- _GPBP17_ expression z-score of over 3 and chromosome 6 short arm (6p) loss
+- _IFT46_ expression z-score of over 3 and chromosome 6 long arm (6q) loss
+- _IFT46_ expression z-score of over 3 and chromosome 6 short arm (6p) loss
+Any tumor with the above molecular characteristics would be exclusively subtyped to the designated group. 
+
+For all other remaining EPN tumors without above molecular characteristics, they would be subtyped to `EPN, ST RELA` and `EPN, ST YAP1` in a non-exclusive way if any of the following alterations were present.
+1. Any tumor with the following alterations would have `EPN, ST RELA` assigned:
```
Comment (jaclyn-taroni) [inline content/03.methods.md:407]:
```suggestion
1. Any tumor with the following alterations was assigned `EPN, ST RELA`: 
```

File: content/03.methods.md [hunk 6]
```diff
@@ -388,6 +388,38 @@ Additional details can be found in the analysis [notebook](https://alexslemonade
...
+- _CXorf67_ expression z-score of over 3 and 1q gain 
+- _TKTL1_ expression z-score of over 3 and 1q gain
+4. Any tumor with the following molecular characterization would be subtyped as `EPN, PF B`:
+- _GPBP17_ expression z-score of over 3 and chromosome 6 long arm (6q) loss
+- _GPBP17_ expression z-score of over 3 and chromosome 6 short arm (6p) loss
+- _IFT46_ expression z-score of over 3 and chromosome 6 long arm (6q) loss
+- _IFT46_ expression z-score of over 3 and chromosome 6 short arm (6p) loss
+Any tumor with the above molecular characteristics would be exclusively subtyped to the designated group. 
+
+For all other remaining EPN tumors without above molecular characteristics, they would be subtyped to `EPN, ST RELA` and `EPN, ST YAP1` in a non-exclusive way if any of the following alterations were present.
```
Comment (jaclyn-taroni) [inline content/03.methods.md:406]:
```suggestion
For all other remaining EPN tumors without above molecular characteristics, they would be subtyped to `EPN, ST RELA` and `EPN, ST YAP1` in a non-exclusive way (e.g., a tumor could have both `EPN, ST RELA` and `EPN, ST YAP1` subtypes) if any of the following alterations were present.
```

File: content/03.methods.md [hunk 7]
```diff
@@ -388,6 +388,38 @@ Additional details can be found in the analysis [notebook](https://alexslemonade
...
+- _RELA_ expression z-score of over 3 
+- _L1CAM_ expression z-score of over 3 
+2. Any tumor with the following alterations would have `EPN, ST YAP1` assigned: 
+- `C11orf95--MAML2` fusion 
+- chromosome 11 long arm (9p) loss 
+- chromosome 11 long arm (9q) gain
+- _ARL4D_ expression z-score of over 3 
+- _CLDN1_ expression z-score of over 3
+Note that the above molecular characteristics did not exclusively subtype EPN tumors. 
+That means, if any tumor has for example `PTEN--TAS2R1` fusion and _ARL4D_ expression z-score of over 3, both `EPN, ST YAP1` and `EPN, ST RELA` would be assigned to the sample.
```
Comment (jaclyn-taroni) [inline content/03.methods.md:420]: Moved a note to the first sentence about this

```suggestion
```

File: content/03.methods.md [hunk 8]
```diff
@@ -388,6 +388,38 @@ Additional details can be found in the analysis [notebook](https://alexslemonade
...
+- _GPBP17_ expression z-score of over 3 and chromosome 6 short arm (6p) loss
+- _IFT46_ expression z-score of over 3 and chromosome 6 long arm (6q) loss
+- _IFT46_ expression z-score of over 3 and chromosome 6 short arm (6p) loss
+Any tumor with the above molecular characteristics would be exclusively subtyped to the designated group. 
+
+For all other remaining EPN tumors without above molecular characteristics, they would be subtyped to `EPN, ST RELA` and `EPN, ST YAP1` in a non-exclusive way if any of the following alterations were present.
+1. Any tumor with the following alterations would have `EPN, ST RELA` assigned: 
+- `PTEN--TAS2R1` fusion 
+- chromosome 9 short arm (9p) loss 
+- chromosome 9 long arm (9q) loss
```
Comment (jaclyn-taroni) [inline content/03.methods.md:410]:
```suggestion
- chromosome 9 arm (9p or 9q) loss 
```

File: content/03.methods.md [hunk 9]
```diff
@@ -388,6 +388,38 @@ Additional details can be found in the analysis [notebook](https://alexslemonade
...
+1. Any tumor with the following alterations would have `EPN, ST RELA` assigned: 
+- `PTEN--TAS2R1` fusion 
+- chromosome 9 short arm (9p) loss 
+- chromosome 9 long arm (9q) loss 
+- _RELA_ expression z-score of over 3 
+- _L1CAM_ expression z-score of over 3 
+2. Any tumor with the following alterations would have `EPN, ST YAP1` assigned: 
+- `C11orf95--MAML2` fusion 
+- chromosome 11 long arm (9p) loss 
+- chromosome 11 long arm (9q) gain
```
Comment (jaclyn-taroni) [inline content/03.methods.md:416]:
```suggestion
- chromosome 11 short arm (11p) loss 
- chromosome 11 long arm (11q) gain
```

**Review by jaclyn-taroni — COMMENTED** [review]
I think this looks good and I agree that `PTEN--TAS2R1` probably should not be in the table you took a screenshot of earlier. Otherwise, it's not clear to me that the _order_ of fusions in the `EPN, ST YAP1` column matters so it may not need to be changed.

However, I'm less familiar with this module compared to the others I've been reviewing, so I'm going to request a review from @jashapiro to be sure that this all looks to be in order.

**Comment by runjin326** [comment]
Thanks @jaclyn-taroni ! I have committed your suggestions now :)

**Review by jashapiro — APPROVED** [review]
@runjin326 I looked at the code in the module, and you are absolutely correct that the [Readme there](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/molecular-subtyping-EPN/README.md#folder-content) is incorrect. The table contents should be just as you describe, based on the code in https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/molecular-subtyping-EPN/03-subgrouping_samples.ipynb. 

So everything here looks good, but we should file a small PR to fix the readme.

**Comment by runjin326** [comment]
> @runjin326 I looked at the code in the module, and you are absolutely correct that the [Readme there](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/molecular-subtyping-EPN/README.md#folder-content) is incorrect. The table contents should be just as you describe, based on the code in https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/molecular-subtyping-EPN/03-subgrouping_samples.ipynb.
> 
> So everything here looks good, but we should file a small PR to fix the readme.

@jashapiro , thanks so much for reviewing this!

**Pull request merged**